common.go

```diff
@@ -0,0 +1,37 @@
+package set
+
+// CommonSet is the interface that all sets implement
+type CommonSet[T any] interface {
+	Slice() []T
+	Insert(T) bool
+	InsertSlice([]T) bool
+	Size() int
+	// ForEach  will call the callback function for each element in the set
+	// If the callback returns false, the iteration will stop
+	ForEach(call func(T) bool)
+}
+
+// InsertSliceFunc inserts all elements from the slice into the set
+func InsertSliceFunc[T, E any](set CommonSet[T], items []E, f func(element E) T) {
```

Comment:
`set CommonSet[T]` - avoid shadowing the package name `set` - we could probably just call this parameter `s` and it would still be clear